File: mobile/ios/CentralOperacionalApp/Models/OperationalSnapshot.swift
import Foundation

/// Payload completo de uma "foto" da operação — o que a API devolve de uma
/// vez (usuário + métricas + módulos), consumido pelo ViewModel a cada
/// atualização em tempo real.
struct OperationalSnapshot: Decodable, Equatable {
    var user: OperationalUser
    var metrics: OperationalMetrics
    var modules: [OperationalModule]

    enum CodingKeys: String, CodingKey {
        case user
        case metrics
        case modules
    }
}

File: mobile/android/app/src/main/java/com/pedidoprime/centraloperacional/model/OperationalSnapshot.kt
package com.pedidoprime.centraloperacional.model

/** Payload completo de uma "foto" da operação — o que o backend devolveria. */
data class OperationalSnapshot(
    val user: OperationalUser,
    val metrics: OperationalMetrics,
    val modules: List<OperationalModule>
)
